lib/emitter.js

```diff
@@ -0,0 +1,28 @@
+const { EventEmitter } = require('events');
+const model = require('./model');
+const { request } = require('./request');
+
+const initEmitter = (config) => {
+  const emitter = new EventEmitter();
+  emitter.on('notify', async (eventName, payload = {}) => {
```

Comment:
Instead of using a single notify event, we should use the actual even names in the config so that a handler function can emit the very same events that they have in config.

Reply:
👍